### DISCUSSION TITLE
Investigate Anglican

### DISCUSSION THREAD

**Discussion by ben18785** [pints-team/pints#1618] [Ideas/Science]
[Key paper](https://probprog.github.io/anglican/index.html) by Frank Wood (at Oxford).

In a way it's similar (although won't have the same users as PINTS as it is much more abstract) but more general. It allows a variety of algorithms under the hood and I've heard it's been used to perform inference on complex blackbox systems.

**Comment by martinjrobins** [comment]
I note that Frank Woods has also done [Probabilistic C](http://proceedings.mlr.press/v32/paige14.pdf), which is a C library (called a "probabilistic programming compilation target") for SMC inference. It has an "observe / predict" structure rather similar to PINTS "ask / tell"
